File: solutions/java/page.java
import com.splunk.*; 
import com.splunk.JobArgs.ExecutionMode;

import java.util.*;
import java.io.*;

public class page {

    public static void main(String[] args) {

		try {
			if(args.length != 1) {
				System.out.println("syntax: setmode \"search query\"");
				System.exit(0);
			}
			Command command = Command.splunk("info").parse(args);
	        Service service = Service.connect(command.opts);

	        JobArgs ja = new JobArgs();
	        ja.setExecutionMode(ExecutionMode.BLOCKING); //forces the call to create() to block until complete
			Job job = service.getJobs().create(args[0], ja);

			int total = job.getResultCount(); 
			int offset = 0;
			int count = 5;
			while (offset < total) {
				System.out.println("====== PAGE " + (offset / count + 1) + " ============" );
				Args params = new Args();
				params.put("count", count);
				params.put("offset", offset);
				ResultsReaderXml set = new ResultsReaderXml(job.getResults(params));
				Event event;       
				while ((event = set.getNextEvent()) != null) {
					System.out.println("------ RESULT " + event.get("_serial") + " -------------------");
					System.out.flush();
					for (String key: event.keySet())     
						System.out.print(key+"="+event.get(key)+"| "); 
					System.out.println("--");
				}
				offset = offset + count; 
				System.in.read();
			}
		} 
    	catch (Exception e)  {
    		if(e.getMessage() != null) {
    			System.err.println("Error message: " + e.getMessage());
    		} else {
    			System.err.println("Error type: " + e.getClass());
    		}
		}
	}
}

File: solutions/python/page.py

import splunklib.client as client
import splunklib.results as results
import utils, sys

opts = utils.parse(sys.argv[1:], {}, ".splunkrc")
if len(sys.argv) != 2:
    print "Syntax: page.py \"search query\""
    sys.exit()
    
try: 
    service = client.connect(**opts.kwargs)

    params = {"exec_mode": "blocking"}    
    job = service.jobs.create(sys.argv[1], **params)
    
    total      = int(job.resultCount)
    offset   = 0;                       
    pageSize = 10;                       
    
    print "Total results = %s" % str(total)

    while (offset < total):
        print "\n===== Page %s ===================================" % str(offset / pageSize + 1)
        pageParams = {"count": pageSize, "offset": offset}
        page = job.results(**pageParams)
        for result in results.ResultsReader(page):
            print "-----RESULT %s --------------------------------------" % result["_serial"]
            for key in result.keys():
                print "%s=%s|" % (key, result[key]),
            print 
        offset += pageSize
        raw_input('Enter to continue:')

except Exception, err:
    sys.stderr.write('Error: %s\n' % str(err))
